libc/signal/handler.c

```diff
@@ -146,13 +149,70 @@ TEST(handler, sighandler_signal_in_signal)
 	TEST_ASSERT_EQUAL_INT(1, sigismember(&oldset, SIGUSR2));
 }
 
+
+static volatile sig_atomic_t got_signal = 0;
+static volatile sig_atomic_t timeout = 0;
+
+static void usr1_handler(int sig)
+{
+	got_signal = sig;
+}
+
+static void alarm_handler(int sig)
+{
+	timeout = 1;
+}
+
+
+TEST(handler, unblock_pending_signal)
+{
+	sigset_t set;
+
+	TEST_ASSERT_EQUAL_INT(0, sigprocmask(0, NULL, &set));
+	TEST_ASSERT_EQUAL_INT(0, sigaddset(&set, SIGUSR1));
+	TEST_ASSERT_EQUAL_INT(0, sigdelset(&set, SIGALRM));
+	TEST_ASSERT_EQUAL_INT(0, sigprocmask(SIG_SETMASK, &set, NULL));
+
+	signal(SIGUSR1, usr1_handler);
```

Comment:
note: could be done with the `sighandler` function already present in this module